diff --git a/node_modules/@zazuko/query-rdf-data-cube/dist/es/index.js b/node_modules/@zazuko/query-rdf-data-cube/dist/es/index.js
index 06113d8..aa8a329 100644
--- a/node_modules/@zazuko/query-rdf-data-cube/dist/es/index.js
+++ b/node_modules/@zazuko/query-rdf-data-cube/dist/es/index.js
@@ -826,11 +826,7 @@ class SparqlFetcher {
         });
     }
     options(body = "") {
-        const options = clone(this.fetchOptions);
-        if (body) {
-            options.body = body;
-        }
-        return options;
+        return {...this.fetchOptions, body};
     }
 }
 /**
diff --git a/node_modules/@zazuko/query-rdf-data-cube/dist/node/index.js b/node_modules/@zazuko/query-rdf-data-cube/dist/node/index.js
index 4a109dc..a9837e4 100644
--- a/node_modules/@zazuko/query-rdf-data-cube/dist/node/index.js
+++ b/node_modules/@zazuko/query-rdf-data-cube/dist/node/index.js
@@ -832,11 +832,7 @@ class SparqlFetcher {
         });
     }
     options(body = "") {
-        const options = clone(this.fetchOptions);
-        if (body) {
-            options.body = body;
-        }
-        return options;
+        return {...this.fetchOptions, body};
     }
 }
 /**
